src/features/groups/components/ContainerCardList/ContainerCard/RenameContainerForm.tsx

```diff
@@ -0,0 +1,118 @@
+import { KeyboardEvent, useEffect, useRef } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { z } from 'zod';
+
+interface IRenameContainerFormProps {
+  /**
+   * The current container name, which a user willing to change from
+   */
+  currentContainerName: string;
+  /**
+   * The state, if true, the rename form is open
+   */
+  isOpen: boolean;
+  /**
+   * The function to change the state of form to false (=close the input field)
+   */
+  onClose: () => void;
+}
+
+export const RenameContainerForm = ({
+  currentContainerName,
+  isOpen,
+  onClose,
+}: IRenameContainerFormProps) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const form = useForm<z.infer<typeof renameContainerFormSchema>>({
+    resolver: zodResolver(renameContainerFormSchema),
+    defaultValues: {
+      containerName: currentContainerName,
+    },
+  });
+
+  const processSubmit: SubmitHandler<RenameContainerInputs> = async (
+    values: RenameContainerInputs,
+  ) => {
+    const { containerName } = values;
+    if (containerName === currentContainerName) return;
+  };
+
+  /**
+   * Handle the key down event.
+   * when the escape key is pressed, close the form without the submission.
+   * @param event - The keyboard event
+   */
+  const handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      onClose();
+    }
+  };
+
+  /**
+   * Clear the input value.
+   */
+  const handleClearInput = () => {
+    form.setValue('containerName', '');
+  };
+
+  /**
+   * Handle the outside click event.
+   * process the form submission and close the form.
+   */
+  const handleOutsideClick = () => {
+    // process the form submission
+    processSubmit(form.getValues());
+    onClose();
+  };
+
+  useEffect(() => {
+    if (!isOpen) form.reset();
+  }, [isOpen, form]);
+
+  useEffect(() => {
+    if (isOpen) {
+      inputRef.current?.focus();
+    }
+  }, [isOpen]);
+
+  return (
+    <Card asChild>
+      <div className="flex gap-2 items-center px-4 py-2">
+        <div className="flex items-center justify-center bg-accent rounded-full min-w-11 h-11">
+          <Icon icon={ContainerIcon} color="black" size={6} />
+        </div>
+        <Form {...form}>
+          <form onSubmit={form.handleSubmit(processSubmit)} className="w-full">
+            <FormField
+              control={form.control}
+              name="containerName"
+              render={({ field }) => (
+                <FormItem className="">
+                  <FormControl>
+                    <SquareTextInput
+                      {...field}
+                      ref={inputRef}
+                      onKeyDown={handleKeyDown}
+                      handleClearInput={handleClearInput}
+                      handleOutsideClick={handleOutsideClick}
+                      className={`${
+                        form.formState.errors.containerName && 'border-danger'
+                      } text-md`}
```

Comment:
Please do not use template literals for classNames. Instead, please utilize `cn()`.

The reason you need to specify `text-md` here is because `text-lg` is defined in `SquareTextInput` as the default value. However, it would be more natural if the default is set to `1rem`, and in that case, you don't even need to specify it because `1rem` is the default font size within `<body>`. 

So, what I want you to do here is to remove the default font size of the `SquareTextInput` and specify `text-lg` only to the group renaming input.